core/src/main/java/hudson/cli/InstallPluginCommand.java

```diff
@@ -75,17 +75,20 @@ protected int run() throws Exception {
         h.checkPermission(PluginManager.UPLOAD_PLUGINS);
         PluginManager pm = h.getPluginManager();
 
+        if (sources.size() > 1 && name != null) {
+            throw new IllegalArgumentException("-name is incompatible with multiple sources");
```

Comment:
If so, there should be a bug in JIRA describing the issue and it's impact (if somebody passes such options)